config/cli.go

```diff
@@ -212,6 +217,36 @@ func CommaSliceFlag(fs *flag.FlagSet, dest *[]string, name string, value []strin
 	})
 }
 
+// handles -foo=value1:10.3,value2:99.9,value3:0.1,value4:100,value5:0
+func CommaWithPctSliceFlag(fs *flag.FlagSet, dest *map[string]float64, name string, value map[string]float64, usage string) {
+	*dest = value
+	fs.Func(name, usage, func(s string) error {
+		elements := strings.Split(s, ",")
+		if len(elements) == 0 || (len(elements) == 1 && elements[0] == "") {
+			*dest = map[string]float64{}
+			return nil
+		}
+		for _, redirect := range elements {
+			elements := strings.Split(redirect, ":")
+			if len(elements) == 2 {
+				percentage, err := strconv.ParseFloat(elements[1], 64)
+				if err != nil || percentage > 100.0 || percentage < 0.0 {
+					return fmt.Errorf("invalid config %s - should be between 0.0 and 100.00", redirect)
+				} else {
+					(*dest)[elements[0]] = percentage
```

Comment:
**nit:** This will actually mutate the map provided as default (`value` arg). If that value is provided as `nil`, this will also panic.

Maybe we want to always initialize the `*dest` on the beginning of this function (aka move line 226 to 224) instead.

Reply:
I think the mutation isn't that problem in this context. I'd need to copy the default map contents to make it mutation-safe... I think we can have simple implementation as `CommaWithPctSliceFlag()` is called only once